superagi/agent/super_agi.py

```diff
@@ -171,8 +194,22 @@ def execute(self, goals: List[str]):
             return "COMPLETE"
         if action.name in tools:
             tool = tools[action.name]
+            print("_________________TESTING__________________")
+            print(action.name)
+            print(action.args)
             try:
                 observation = tool.execute(action.args)
+                print("Tool Observation : ")
+                print(observation)
+                if action.name == WRITE_FILE:
```

Comment:
Why is the tool code here. Add this inside tool.

Reply:
after writing file to store in db it requires project id which is available here but if we shift this code to write tool under tools, project id would be a param and params of tool are generated by agent so project id can't be there so what to do?